app/models/student_task.rb

```
@@ -1,50 +1,206 @@
+    create_from_participant(AssignmentParticipant.find_by(id: id))
+  end
+
+  # Builds a unified timeline by merging due dates and actual activity,
+  # sorted chronologically — mirrors the old get_timeline_data logic.
+  def self.get_timeline_data(assignment, participant)
+    timeline = []
+
+    # 1. Due dates — labeled as "X deadline" mirroring old get_due_date_data behavior
+    DueDate.fetch_due_dates(assignment).each do |due_date|
+      timeline << {
+        'id'    => nil,
```

RuboCop (1.87.0)

[convention] Align the keys of a hash literal if they span more than one line.

(Layout/HashAlignment)